snappy-core/src/main/scala/org/apache/spark/sql/execution/datasources/StoreDataSourceStrategy.scala

```diff
@@ -77,9 +77,11 @@ private[sql] object StoreDataSourceStrategy extends Strategy with Logging {
       filterPredicates,
       numPartition,
       partitionColumns,
-      (requestedColumns, pushedFilters) => {
-        scanBuilder(requestedColumns, selectFilters(pushedFilters).toArray)
+      (requestedColumns, _, pushedFilters) => {
+        scanBuilder(requestedColumns, pushedFilters.toArray)
       })
+
```

Comment:
@rishitesh Ok. Btw, I also noticed this in ColumnFormatRelation:

```
  override def partitionColumns: Seq[String] = {
    connectionType match {
      case ConnectionType.Embedded => partitioningColumns
      case _ =>   Seq.empty[String] // Temporary fix till we fix Non-EMbededed join
    }
  }
```

Is this issue being tracked somewhere in JIRA? I believe the external shell mode will become important going forward (already we have multiple prospectives who want to run with stock spark in shell mode).